draft-ietf-tls-rfc8446bis.md

```diff
@@ -3517,6 +3517,16 @@ a KeyUpdate with the old key is received before accepting any messages
 encrypted with the new key. Failure to do so may allow message truncation
 attacks.
 
+With a 128-bit key as in AES-128, rekeying 2^64 times has a high
+probability of key reuse within a given connection.  Note that even
+if the key repeats, the IV is also independently generated.  In order
```

Comment:
So you start by saying that rekeying has a high collision chance, but then you say that it doesn't because the nonce also has a random starting point and therefore expands the space.  This is confusing.

There are two analyses you might use here:

1. The key analysis (the first sentence), which is pretty simple.  Simplistic even.
2. The joint analysis of key + nonce, which is 224 bits wide, which the note implies.


This latter one needs to consider the number of records in each epoch as well.  If the per-epoch record limit is 2^25, then assuming a purely random key + a sequential nonce (which is not true but it should work for the purposes of rough analysis), then you can treat the space as drawing from a 199 bit space at random, so you really only need to worry about 2^99 key updates.  Rekeying only 2^48 times retains an advantage in excess of 2^50 there, so maybe that's a good approach.  And should that ever happen, nonces will be used far less (2^48 records is an unlikely number of records for a single connection, let alone 2^72.5), so the odds of collision are probably much lower still.

So I would probably drop this note, except to *maybe* say - at the end - while the risk of collision for a key is low, the use of random base nonces further reduces the risk of collision for record protection, even where keys do happen to be reused.

Reply:
Well, I don't think this text is incorrect. It says that there is a high chance of key collision which there is, so this is correct.